providers/task.rb

```diff
@@ -231,6 +234,49 @@ def validate_create_day
   end
 end
 
+def validate_create_months
+  return unless @new_resource.months
+  unless [:monthly].include?(@new_resource.frequency)
+    fail 'months attribute is only valid for tasks that run weekly or monthly'
+  end
+  months = @new_resource.months.split(',')
+  months.each do |month|
+    unless ['JAN', 'FEB', 'MAR', 'APR', 'MAY', 'JUN', 'JUL', 'AUG', 'SEP', 'OCT', 'NOV', 'DEC', '*'].include?(month.upcase)
+      fail 'months attribute invalid. Only valid values are: JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC and *. Multiple values must be separated by a comma.'
```

Comment:
I can certainly do that, but right now it's consistent with the rest of the file and follows same format as "def validate_create_day". Maybe another revision could replace them both with arrays.